packages/website/src/examples/form/button/ButtonWithPromiseHandler.demo.tsx

```diff
@@ -0,0 +1,18 @@
+import {Button, showNotification} from '@coveord/plasma-mantine';
+
+const promise = () =>
+    new Promise<void>((r) =>
+        setTimeout(
+            () =>
+                r(
+                    showNotification({
+                        title: 'Promise resolved',
+                        message: 'The callback passed to the onClick prop is returning a promise.',
+                        autoClose: false,
+                    })
+                ),
+            3000
+        )
+    );
+
+export default () => <Button onClick={promise}>Async handler</Button>;
```

Comment:
I find the less we mention the word "promise", the clearer it gets for junior UI devs. Also, by extracting the "faking stuff" part out of the click handler we typically would see it makes it more obvisous that the timeout stuff is just as an example.

```suggestion
const somethingAsync = () => new Promise<void>((resolve) => {
    setTimeout(resolve, 3000);
});

const handleClick = async () => {
    await somethingAsync();
    showNotification({
        title: 'Saved successfully',
        message: 'The save disabled was put in a loading state while it was waiting for the save to resolve.',
        autoClose: false,
    });
};

export default () => <Button onClick={handleClick}>Save</Button>;
```

Reply:
Fair enough, I changed the filename to `ButtonWithAsyncLoader.demo.tsx` and updated the demo code